File: Controller/GraphController.cs
using Isocline.Nodes;
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Isocline.Controller
{
    internal class GraphController
    {
        public Image CreateGraphImage(double min, double max, int width, int height, Func<double, double, double> f,double x0,double y0)
        {
            var gm = new GraphMaster(min, max);
            return gm.CreateGraph(width, height,
                    (g) => gm.DrawLinearInterpolation(g, f, x0, y0));
        }
    }
}

File: Controller/GraphController.vb
Imports System
Imports System.Drawing

Namespace Isocline.Controller
    Friend Class GraphController
        Public Function CreateGraphImage(ByVal min As Double, ByVal max As Double, ByVal width As Integer, ByVal height As Integer, ByVal f As Func(Of Double, Double, Double), ByVal x0 As Double, ByVal y0 As Double) As Image
            Dim gm = New GraphMaster(min, max)
            Return gm.CreateGraph(width, height, Sub(g) gm.DrawSolution(g, f, x0, y0))
        End Function
    End Class
End Namespace
